### PR TITLE
Add IDL for governance program

### PR DESCRIPTION
**Pull request by gagliardetto**
This is a draft of an IDL for the governance program.

NOTES:
----------

- There might be some things that are in `accounts` and should go to `types` instead (and viceversa).
- Instruction accounts are generated from the comments (such as `// 0. `[writable]` The account to reallocate.`)
- Names of accounts could be improved (maybe too long, or not clear enough).

### CODE REVIEW

File: governance/idl.json [hunk 1]
```diff
@@ -0,0 +1,4865 @@
...
+						]
+					}
+				]
+			},
+			"docs": [
+				"Program metadata account. It stores information about the particular SPL-Governance program instance"
+			]
+		},
+		{
+			"name": "ProposalOption",
```
Comment (ngundotra) [inline governance/idl.json:2648]: This should be in `types` not `accounts`

**Comment by ngundotra** [comment]
@gagliardetto this looks great! Missing a few new accounts however:

- ProgramGovernanceV2
- MintGovernanceV2
- TokenGovernanceV2
- SignatoryRecordV2

I think it may be better to go with auto generating the IDL using something like [MPL's shank](https://github.com/metaplex-foundation/shank)

**Comment by ngundotra** [comment]
Might be helpful to just annotate using Shank once this PR is in: https://github.com/metaplex-foundation/shank/pull/32

**Comment by gagliardetto** [comment]
> Might be helpful to just annotate using Shank

Yep! Was thinking the same. It would avoid needing ad-hoc updates to the IDL json file, and remove the need to mentally understand what/how to update.

**Pull request closed without merging**